src/OptionSettlement.sol

```diff
@@ -108,14 +109,38 @@ contract OptionSettlementEngine is ERC1155, IOptionSettlementEngine {
         override
         returns (string memory)
     {
-        if (tokenType[tokenId] != Type.None) {
+        Type _type = tokenType[tokenId];
+        Option memory optionInfo;
+
+        if (_type == Type.None) {
             revert TokenNotFound(tokenId);
+        } else if (_type == Type.Claim) {
+            Claim memory claimInfo = _claim[tokenId];
+
+            optionInfo = _option[claimInfo.option];
+        } else {
+            optionInfo = _option[tokenId];
         }
-        // TODO(Implement metadata/uri with frontend dev)
-        string memory json = Base64.encode(
-            bytes(string(abi.encodePacked("{}")))
+
+        string memory svg = NFTGenerator.generateNFT(
+            NFTGenerator.GenerateNFTParams({
+                underlyingAsset: optionInfo.underlyingAsset,
+                exerciseAsset: optionInfo.exerciseAsset,
+                exerciseTimestamp: optionInfo.exerciseTimestamp,
+                expiryTimestamp: optionInfo.expiryTimestamp,
+                underlyingAmount: optionInfo.underlyingAmount,
+                exerciseAmount: optionInfo.exerciseAmount,
+                tokenType: _type
+            })
         );
-        return string(abi.encodePacked("data:application/json;base64,", json));
+
+        return
+            string(
+                abi.encodePacked(
```

Comment:
https://eips.ethereum.org/EIPS/eip-1155#metadata

"The URI MUST point to a JSON file that conforms to the "ERC-1155 Metadata URI JSON Schema"

```{
	"name": "Asset Name",
	"description": "Lorem ipsum...",
	"image": "https:\/\/s3.amazonaws.com\/your-bucket\/images\/{id}.png",
	"properties": {
		"simple_property": "example value",
		"rich_property": {
			"name": "Name",
			"value": "123",
			"display_value": "123 Example Value",
			"class": "emphasis",
			"css": {
				"color": "#ffffff",
				"font-weight": "bold",
				"text-decoration": "underline"
			}
		},
		"array_property": {
			"name": "Name",
			"value": [1,2,3,4],
			"class": "emphasis"
		}
	}
}```

We need to wrap the base64 encoded svg in a json file containing other metadata about the token here.

In the case of claims, it makes sense to have the SVG, in the case of options, not so sure. I guess it does, just because it looks cool, but an options balance may be greater than 1.

Reply:
Ok this is updated. The resulting encoded json will look like this:

```
{
  "name": "WETHDAI220826C",
  "description": "NFT representing a Valorem options contract. WETH Address: 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2. DAI Address: 0x6b175474e89094c44da98b954eedeac495271d0f.",
  "image": "data:image/svg+xml;base64,PHN2ZyB3aWR0aD0nNDAwJyBoZWlnaHQ9JzMwMCcgdmlld0JveD0nMCAwIDQwMCAzMDAnIHhtbG5zPSdodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2Zyc+PHJlY3Qgd2lkdGg9JzEwMCUnIGhlaWdodD0nMTAwJScgcng9JzEyJyByeT0nMTInICBmaWxsPScjM0U1REM3JyAvPjxnIHRyYW5zZm9ybT0nc2NhbGUoNSksIHRyYW5zbGF0ZSgyNSwgMTgpJyBmaWxsLW9wYWNpdHk9JzAuMTUnPjxwYXRoIHhtbG5zPSdodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZycgZD0nTTY5LjM1NzcgMTQuNTAzMUgyOS43MjY1TDM5LjYzMTIgMEgwTDE5LjgxNTYgMjlMMjkuNzI2NSAxNC41MDMxTDM5LjYzMTIgMjlIMTkuODE1NkgwTDE5LjgxNTYgNThMMzkuNjMxMiAyOUw0OS41NDIxIDQzLjUwMzFMNjkuMzU3NyAxNC41MDMxWicgZmlsbD0nd2hpdGUnLz48L2c+PHRleHQgeD0nMTZweCcgeT0nNTVweCcgZm9udC1zaXplPSczMnB4JyBmaWxsPScjZmZmJyBmb250LWZhbWlseT0nSGVsdmV0aWNhJz5XRVRIIC8gREFJPC90ZXh0Pjx0ZXh0IHg9JzE2cHgnIHk9JzgwcHgnIGZvbnQtc2l6ZT0nMTYnIGZpbGw9JyNmZmYnIGZvbnQtZmFtaWx5PSdIZWx2ZXRpY2EnIGZvbnQtd2VpZ2h0PSczMDAnPkxvbmcgQ2FsbDwvdGV4dD48dGV4dCB4PScxNnB4JyB5PScxMTZweCcgZm9udC1zaXplPScxNCcgbGV0dGVyLXNwYWNpbmc9JzAuMDFlbScgZmlsbD0nI2ZmZicgZm9udC1mYW1pbHk9J0hlbHZldGljYSc+VU5ERVJMWUlORyBBU1NFVDwvdGV4dD48dGV4dCB4PScxNnB4JyB5PScxNDBweCcgZm9udC1zaXplPScxOCcgZmlsbD0nI2ZmZicgZm9udC1mYW1pbHk9J0hlbHZldGljYScgZm9udC13ZWlnaHQ9JzMwMCc+NyBXRVRIPC90ZXh0Pjx0ZXh0IHg9JzE2cHgnIHk9JzE3NnB4JyBmb250LXNpemU9JzE0JyBsZXR0ZXItc3BhY2luZz0nMC4wMWVtJyBmaWxsPScjZmZmJyBmb250LWZhbWlseT0nSGVsdmV0aWNhJz5FWEVSQ0lTRSBBU1NFVDwvdGV4dD48dGV4dCB4PScxNnB4JyB5PScyMDBweCcgZm9udC1zaXplPScxOCcgZmlsbD0nI2ZmZicgZm9udC1mYW1pbHk9J0hlbHZldGljYScgZm9udC13ZWlnaHQ9JzMwMCc+MzAwMCBEQUk8L3RleHQ+PHRleHQgeD0nMTZweCcgeT0nMjM2cHgnIGZvbnQtc2l6ZT0nMTQnIGxldHRlci1zcGFjaW5nPScwLjAxZW0nIGZpbGw9JyNmZmYnIGZvbnQtZmFtaWx5PSdIZWx2ZXRpY2EnPkVYRVJDSVNFIERBVEU8L3RleHQ+PHRleHQgeD0nMTZweCcgeT0nMjYwcHgnIGZvbnQtc2l6ZT0nMTgnIGZpbGw9JyNmZmYnIGZvbnQtZmFtaWx5PSdIZWx2ZXRpY2EnIGZvbnQtd2VpZ2h0PSczMDAnPjA4LzI1LzIwMjI8L3RleHQ+PHRleHQgeD0nMjAwcHgnIHk9JzIzNnB4JyBmb250LXNpemU9JzE0JyBsZXR0ZXItc3BhY2luZz0nMC4wMWVtJyBmaWxsPScjZmZmJyBmb250LWZhbWlseT0nSGVsdmV0aWNhJz5FWFBJUlkgREFURTwvdGV4dD48dGV4dCB4PScyMDBweCcgeT0nMjYwcHgnIGZvbnQtc2l6ZT0nMTgnIGZpbGw9JyNmZmYnIGZvbnQtZmFtaWx5PSdIZWx2ZXRpY2EnIGZvbnQtd2VpZ2h0PSczMDAnPjA4LzI2LzIwMjI8L3RleHQ+PC9zdmc+"
}
```